root/static/scripts/edit/URLCleanup.js

```diff
@@ -388,7 +408,7 @@ const CLEANUPS = {
     },
   },
   'amazon': {
-    match: [new RegExp('^(https?://)?([^/]+\\.)?(amazon\\.(com|ca|co\\.uk|fr|at|de|it|co\\.jp|jp|cn|es|in|com\\.br|com\\.mx|com\\.au)|amzn\\.com)', 'i')],
+    match: [new RegExp('^(https?://)?((?!music)[^/]+\.)?(amazon\\.(com|ca|co\\.uk|fr|at|de|it|co\\.jp|jp|cn|es|in|com\\.br|com\\.mx|com\\.au)|amzn\\.com)', 'i')],
```

Comment:
As a follow-up to https://github.com/metabrainz/musicbrainz-server/pull/1189#discussion_r331579261, here is an alternative that works with `anysubdomainto.music.amazon.co.jp` but also dismisses `anysubdomainwithmusicinside.amazon.co.jp`:
```suggestion
    match: [new RegExp('^(https?://)?(((?!music)[^/])+\.)?(amazon\\.(com|ca|co\\.uk|fr|at|de|it|co\\.jp|jp|cn|es|in|com\\.br|com\\.mx|com\\.au)|amzn\\.com)', 'i')],
```